### ISSUE TITLE
super_editor is incompatible with google_fonts due to http version

### ISSUE THREAD

**Issue by milindgoel15** [superlistapp/super_editor#1464]
Because every version of super_editor depends on http ^0.13.1 and google_fonts >=4.0.5 depends on http ^1.0.0, super_editor is incompatible with google_fonts >=4.0.5.
So, because todoify depends on both google_fonts ^6.0.0 and super_editor ^0.2.6, version solving failed.
exit code 1

I checked out the super_editor in code and the version constraint is still 0.13.1. Could you please upgrade it?

**Comment by rutvik110** [comment]
@milindgoel15 `super_editor` specifies the following range for `http, `>=0.13.1 <2.0.0` which in my understanding should be resolved by `Pub` to a version equal to or greater than `v1.0.0`  which is required by `google_fonts`.  Right now, I'm not sure why this isn't picked up by the package manager and resulting in the given error. For the moment you can override the `http` deps like follow, 
 ```
 dependency_overrides:
  http: ^1.0.0
 ```
 
 @matthew-carroll Could there any reason as to why package manager won't satisfy this constraints and resolve the `http` version to next major release for `super_editor` required for `google_fonts`? As per my understanding, our constraints should allow usage of `google_fonts` which depends on http `^1.0.0`.

**Comment by matthew-carroll** [comment]
@rutvik110 I'm not sure why this would happen.

@milindgoel15 can you please provide a way for us to reproduce this issue on our side?

**Comment by jellynoone** [comment]
I think the issue here is that @rutvik110 is using the most recently published version of the package on pub.dev, `0.2.6`, which constrains the `http` package to `^0.13.1`.

Attaching the relevant parts of `pubspec.yaml` from tag `0.2.6`

https://github.com/superlistapp/super_editor/blob/3887ee6d2ff94ef99f05d71db88577d4c6a63eca/super_editor/pubspec.yaml#L26-L28

**Comment by rutvik110** [comment]
ah okay. I assumed that the fix from an earlier issue #1206 that updated `http` version was pushed to pub.dev. 

In that case, @milindgoel15 till we get the next release out which has this fix, you can override the http deps in `dependency_overrides` to `1.0.0` as required. There aren't any compatibility issues between the `0.13.1` and `1.0.0` as discussed in the mentioned PR so you should be good to go.

**Comment by rutvik110** [comment]
I'm closing this issue as the fix for this has already landed on `main` with #1206 and should be contained in next release for pub.dev. 

Till the next release, overriding the `http` deps in `pubspec.yaml` like follow should resolve the issue,

```yaml
dependency_overrides:
 http: ^1.0.0
```

**Comment by milindgoel15** [comment]
Another dependency version error:

Resolving dependencies...
Because every version of super_editor depends on linkify ^4.0.0 and flutter_linkify 6.0.0 depends on linkify ^5.0.0, super_editor is incompatible with flutter_linkify 6.0.0.
And because no versions of flutter_linkify match >6.0.0 <7.0.0, super_editor is incompatible with flutter_linkify ^6.0.0.
So, because todoify depends on both flutter_linkify ^6.0.0 and super_editor ^0.2.6, version solving failed.


You can try the following suggestion to make the pubspec resolve:
* Consider downgrading your constraint on flutter_linkify: flutter pub add flutter_linkify:^5.0.2
